controller/volume_controller.go

```diff
@@ -4074,12 +4078,14 @@ func (c *VolumeController) IsReplicaUnavailable(r *longhorn.Replica) (bool, erro
 	if r.Spec.NodeID == "" || r.Spec.DiskID == "" || r.Spec.DiskPath == "" || r.Spec.DataDirectoryName == "" {
 		return true, nil
 	}
+	log := c.logger.WithField("replica", r.Name)
 
 	isDownOrDeleted, err := c.ds.IsNodeDownOrDeleted(r.Spec.NodeID)
 	if err != nil {
 		return true, errors.Wrapf(err, "failed  to check if node %v is still running for failed replica %v", r.Spec.NodeID, r.Name)
 	}
 	if isDownOrDeleted {
+		log.Warnf("Replica is unavailable due to Node %v is NotReady or Deleted", r.Spec.NodeID)
```

Comment:
The same, suggest logging at the caller. If you need a specific reason when the bool var is true, leverage the returned error or even a third reason returned variable.

Reply:
Thanks for the reminder, I would like to wrap the error and return it.